block-producer/src/batch_builder/mod.rs

```diff
@@ -18,66 +25,105 @@ mod tests;
 ///
 /// Note: Until recursive proofs are available in the Miden VM, we don't include the common proof.
 pub struct TransactionBatch {
-    txs: Vec<SharedProvenTx>,
+    account_initial_states: Vec<(AccountId, Digest)>,
+    updated_accounts: Vec<(AccountId, Digest)>,
```

Comment:
Would it make sense to combine these into a single data structure (`BTreeMap`) similar to how we did it in the other place?

Reply:
done in [7bbef20](https://github.com/0xPolygonMiden/miden-node/pull/64/commits/7bbef20852d633e84574e1f34f3e04d040580978)